torchtune/utils/_distributed.py

```diff
@@ -212,3 +237,60 @@ def lora_wrap_fsdp(module: nn.Module, recurse: bool, **kwargs):
         return isinstance(module, tuple(modules_to_wrap))
 
     return lora_wrap_fsdp
+
+
+def get_full_finetune_fsdp_wrap_policy(
+    memory_efficient_fsdp_wrap: bool, modules_to_wrap: Set[Type]
+) -> FSDPPolicyType:
+    """
+    Retrieves an FSDP wrapping policy based on the specified ``model_type`` and ``modules_to_wrap``.
+    Specifically, for Llama3 model variants, the policy returned will wrap the model's token embedding
+    and output projection in addition to the modules specified in ``modules_to_wrap`` to maximize memory savings.
+    For all other models, only the modules specified will be wrapped.
+    Args:
+        memory_efficient_fsdp_wrap (bool): If ``True``, will also wrap embedding and output projection layers
+        with FSDP.
+        modules_to_wrap (Set[Type]): Set of module types to wrap.
+
+    Note:
+        ``memory_efficient_fsdp_wrap`` memory improvements have currently only been verified on llama3 workloads,
+            to provide ~15% memory improvement (when used alongside AC memory efficient wrapping). Other workloads
+            have not been verified and may not see the same improvements.
+    Returns:
+        FSDPPolicyType: Wrapping policy that can be passed into ``FullyShardedDataParallel`` as the ``auto_wrap_policy``
+            argument. Please see documentation for `torchtune.utils.FSDPPolicyType` for additional details.
```

Comment:
Is there any way to link directly to this docstr? @ebsmothers or @NicolasHug  maybe?